tests_e2e/orchestrator/lib/agent_test_loader.py

```diff
@@ -15,77 +15,178 @@
 # limitations under the License.
 #
 import importlib.util
-import json
+# E0401: Unable to import 'yaml' (import-error)
+import yaml  # pylint: disable=E0401
 
 from pathlib import Path
 from typing import Any, Dict, List, Type
 
+import tests_e2e
 from tests_e2e.tests.lib.agent_test import AgentTest
 
 
-class TestSuiteDescription(object):
+class TestSuiteInfo(object):
     """
-    Description of the test suite loaded from its JSON file.
+    Description of a test suite
     """
+    # The name of the test suite
     name: str
+    # The tests that comprise the suite
     tests: List[Type[AgentTest]]
+    # An image or image set (as defined in images.yml) specifying the images the suite must run on.
+    images: str
```

Comment:
the solution is the same as in dcr: create a new image set that includes all the images that the suite needs to run on and use that set

Reply:
In DCR, we create the new image set in specific test_suite, now we need to define all those in image.yml. If we end of doing it in one file, it becomes too much in single file. May be worth considering allowing new image set at test_suite level. For now, ok and I guess when we port more tests, we get the idea whether we need to implement this or not.

Reply:
Yes, start simple and add functionality only if actually needed. Using an array instead of a single item is pretty simple, actually, but for the moment keep things simple.